mohawk/util.py

```diff
@@ -268,3 +275,110 @@ def normalize_header_attr(val):
     # Normalize like the hawk reference code.
     val = escape_header_attr(val)
     return val
+
+
+def get_bewit(resource):
+    """Returns a bewit identifier for the resource as a string"""
+    if resource.method != 'GET':
+        raise ValueError('bewits can only be generated for GET requests')
+    mac = calculate_mac(
+        'bewit',
+        resource,
+        None,
+    )
+
+    if isinstance(mac, six.binary_type):
+        mac = mac.decode('ascii')
+
+    if resource.ext is None:
+        ext = ''
+    else:
+        ext = resource.ext
+
+    # Strip out \ from the clientId
+    # since that can break parsing the response
+    # NB that the canonical implementation does not do this as of
+    # Oct 28, 2015, so this could break compat.
+    # We can leave \ in ext since validators can limit how many \ they split
+    # on (although again, the canonical implementation does not do this)
+    clientId = resource.credentials['id'].replace("\\", "")
+
+    # b64encode works only with bytes in python3, but all of our parameters are
+    # in unicode, so we need to encode them. The cleanest way to do this that
+    # works in both python 2 and 3 is to use string formatting to get a
+    # unicode string, and then explicitly encode it to bytes.
+    inner_bewit = u"{id}\\{exp}\\{mac}\\{ext}".format(
+        id=clientId,
+        exp=resource.timestamp,
+        mac=mac,
+        ext=ext,
+    )
+    inner_bewit_bytes = inner_bewit.encode('ascii')
+    bewit_bytes = urlsafe_b64encode(inner_bewit_bytes)
+    # Now decode the resulting bytes back to a unicode string
+    return bewit_bytes.decode('ascii')
+
+
+bewittuple = namedtuple('bewittuple', 'id exp mac ext')
+
+def parse_bewit(bewit):
+    """Returns the parts of the bewit as a namedtuple:
+        (id, exp, mac, ext)
+    Input it base64-encoded bewit string
+    """
+    decoded_bewit = b64decode(bewit).decode('ascii')
+    return bewittuple(*decoded_bewit.split("\\", 3))
```

Comment:
This will raise a `TypeError` if the bewit has no slashes at all or has the wrong number. You should catch it instead and raise a new error like `InvalidBewit` which could go in [mohawk.exc](https://github.com/catlee/mohawk/blob/bewit/mohawk/exc.py). At the very least, you need some tests for parsing errors.